### PR TITLE
Add doxygen warning checks.

### PR DESCRIPTION
**Pull request by sarenameas**
### The Doxygen warning check do these steps:
- Recursively clone this repo.
- Run the Doxygen command on each of the library spoke repos. The warnings for the libraries are already checked in the library repo's themselves, so there is no need to check the output.
- Run the Doxygen command on this repo. Check the output for warnings. 

### Test:
- See commit 1a2b8db for the test that adds a warning in the CSDK doxygen. The check fails.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

**Review by nateglims — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -114,3 +114,27 @@ jobs:
...
+        run: |
+          wget -qO- "http://doxygen.nl/files/doxygen-1.8.20.linux.bin.tar.gz" | sudo tar --strip-components=1 -xz -C /usr/local
+          sudo apt-get install -y libclang-9-dev
+      - name: Run Doxygen on all library submodules
+        run: |
+          ROOT=$PWD
+          for lib in $(find libraries/aws libraries/standard -maxdepth 1 -mindepth 1)
+          do
+            cd $lib
+            if [ ! -d docs/doxygen/ ]; then exit 0; fi
```
Comment (gkwicker) [inline .github/workflows/ci.yml:133]: Should we log a message if there is nothing in `docs/doxygen`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -114,3 +114,27 @@ jobs:
...
+            doxygen docs/doxygen/config.doxyfile
+            cd $ROOT
+          done
+      - name: Run Doxygen on CSDK and verify stdout is empty
+        run: |
+          doxygen docs/doxygen/config.doxyfile 2>&1 | tee doxyoutput.txt
+          if [[ "$(wc -c < doxyoutput.txt | bc)" = "0" ]]; then exit 0; else exit 1; fi
```
Comment (gkwicker) [inline .github/workflows/ci.yml:140]: If `doxyoutput.txt` isn't empty, do we want to log it so that the developer can see the errors?

**Review by gkwicker — DISMISSED** [review]
Approved, but there are a couple of questions in the comments.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sarenameas) [reply .github/workflows/ci.yml:140]: yes those are shown with tee above (which outputs to both the stdout and the text file).

Comment (sarenameas) [reply .github/workflows/ci.yml:140]: image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sarenameas) [reply .github/workflows/ci.yml:133]: I can see that being useful I will add that

**Review by gkwicker — APPROVED** [review]

**Review by nateglims — APPROVED** [review]

**Pull request merged**